File: tutorial/filter.rb
# extended Array
class Array
  def odd_number
    select(&:odd?)
  end

  def times_of(n)
    select { |a| a % n == 0 }
  end
end

ary = 1.upto(20).to_a
# => [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
ary.odd_number.times_of(3)
# => [3, 9, 15]

File: tutorial/filter.py
class MyList(list):
    def odd_number(self):
        return List(filter(lambda a: a % 2 == 1, self))

    def times_of(self, n):
        return List(filter(lambda a: a % n == 0, self))

a = List(range(1, 21))
a.odd_number().times_of(3)
